fsspec/utils.py

```diff
@@ -266,11 +295,11 @@ def read_block(f, offset, length, delimiter=None, split_before=False):
         length = end - start
 
     f.seek(offset)
-    b = f.read(length)
+    b = f.read(length)  # type: ignore[arg-type]
```

Comment:
Ma even go further and raise a ValueError is length is None and delimiter is None that they cannot both be None at the same time.

Reply:
I added an `assert` here, but I wonder if we should to what I originally suggested of:
```
if length is None:
    b = f.read()
else:
    b = f.read(length)
```

The docs say that if `length is None` we want to read until the end of the string, which I believe this will.

>>> unless f.read can actually take None and it's just a type stubbing problem?

Arguably it is? I did just test with an actual file and confirm that `open(file).read(None)` gives reasonable results.